src/ui/FacetValueSuggestions/FacetValueSuggestions.ts

```
@@ -0,0 +1,350 @@
+   */
+  constructor(element: HTMLElement, public options: IFacetValueSuggestionsOptions, bindings?: IComponentBindings) {
+    super(element, FacetValueSuggestions.ID, bindings);
+
+    this.options = ComponentOptions.initComponentOptions(element, FacetValueSuggestions, options);
+
+    Assert.check(Utils.isCoveoField(<string>this.options.field), `${this.options.field} is not a valid field`);
+
+    this.queryStateFieldFacetId = `f:${this.options.field}`;
+    this.queryStateFieldValueId = `fv:${this.options.field}`;
+    if (this.queryStateModel.getAttributes()[this.queryStateFieldValueId] === undefined) {
+      this.queryStateModel.registerNewAttribute(this.queryStateFieldValueId, []);
+    }
+
+    $$(this.root).on(OmniboxEvents.populateOmniboxSuggestions, (e: Event, args: IPopulateOmniboxSuggestionsEventArgs) => {
+      args.suggestions.push(this.getSuggestion(args.omnibox));
+    });
+  }
+
+  public getSuggestion(omnibox: Omnibox): Promise<IOmniboxSuggestion[]> {
+    const text = omnibox.getText();
+
+    return this.cache.getSuggestions(text, () => this.getFacetValueSuggestions(text, omnibox));
+  }
+
+  private getFacetValueSuggestions(text: string, omnibox: Omnibox): Promise<IOmniboxSuggestion[]> {
+    const wordsToQuery = this.options.useValueFromSearchbox ? [text] : [];
+
+    const promise: Promise<IOmniboxSuggestion[]> = this.getQuerySuggestionsKeywords(omnibox)
+      .then(suggestions => {
+        return this.listFieldValuesBatch(_.unique(wordsToQuery.concat(suggestions)));
+      })
+      .then((response: IFacetValueSuggestionsResponse) => {
+        return this.filterCurrentlySelectedValuesFromResults(response.responses, response.reference).map(resultToShow => {
+          return <IOmniboxSuggestion>{
+            html: this.buildDisplayNameForRow(resultToShow, omnibox),
+            onSelect: () => this.onRowSelection(resultToShow, omnibox)
+          };
+        });
+      })
+      .catch(() => []);
```

Might want to not silently fail here. At least logging something in the console would be helpful.